### ISSUE TITLE
Provide valid config.json's and more docs/guidance for each of the three user project area hardening options

### ISSUE THREAD

**Issue by grayresearch** [efabless/caravel_user_project#237]
I am preparing a challenging,  dense embedded FPGA user_project for MPW-8.

Referring to the three hardening options (please give them concise names we can all use)
https://github.com/efabless/caravel_user_project/blob/main/docs/source/index.rst#hardening-the-user-project-using-openlane
I would like to use option 3. This links to example caravel_ibex. Unfortunately the latter uses the deprecated config.tcl.

Then contrasting
https://github.com/efabless/caravel_user_project/blob/main/openlane/user_project_wrapper/config.json
vs.
https://github.com/efabless/caravel_ibex/blob/master/openlane/user_project_wrapper/config.tcl
it is difficult to know which aspects of the former must be preserved in a json-ification of the latter, or should the user_project_wrapper's config.json be ignored and instead just json'ify the caravel_ibex's config.tcl options.

Also I see from caravel_ibex's
set ::env(GLB_RT_MAXLAYER) 5
that an "option 3" top-level user project can use the extra metal layer (although its subordinate macros cannot). So I infer that an "option 2" user project (which seems to be just a macro-less "option 3") can also use GLB_RT_MAXLAYER 5. If so, this is an important consideration in favor of option 2/3 for a routing-challenging full user project. Please document this explicitly -- option 2/3 can/cannot use layer 5.

In summary: working, recommended config.json's and some more documentation on the three hardening options would help new designers make best use of the caravel platform. Thank you.
